### ISSUE TITLE
Check that buffer is correctly accessed in WHERE clause

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#1074]
I activated this rule today, it seems to be checking the underlying table and not the actual buffer being used:
````
block-level on error undo, throw.

define buffer customer for customer.
define buffer bucustomer for customer.

find first customer no-lock no-error.
for each bucustomer
   where bucustomer.custnum > 0
   and   customer.custnum > 0
no-lock:
   // false positive on buffer usage customer.custnum > 0
end.
````

**Comment by gquerret** [comment]
Reproduced. Investigating root cause.
